### PR TITLE
Fix PHP notice that showed when orderby query parameter was an array (With Unit tests)

### PR DESCRIPTION
**Pull request by Rajinsharwar**
This is a followup to the https://github.com/WordPress/wordpress-develop/pull/5404, with unit tests.

Trac ticket: https://core.trac.wordpress.org/ticket/59494

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

**Comment by swissspidy** [comment]
Using `wp_remote_*` is not really the right approach here. If you want to test the frontend output like this, you would write an e2e test. However, this change here does not really warrant an e2e test. A simple unit test that tests the method is good enough. Doing an HTTP call is not really a unit test.

Look at the other unit tests in `tests/phpunit/tests/admin/wpListTable.php` and how they're written.

You'll want a simple test that calls the `search_box()` method and verifies the output is as expected.

The structure could look like this:

```php
	/**
	 * Tests that `WP_List_Table::search_box()` supports multiple orderby fields.
	 *
	 * @ticket 59494
	 *
	 * @covers WP_List_Table::search_box
	 */
	public function test_search_box_supports_multiple_orderby() {
		$_REQUEST['s'] = 'search term';

		$output = get_echo( array( $this->list_table, 'search_box' ), array( 'foo', 'bar' ) );

		$this->assertStringContainsString( 'Write your expected string here', $output );
	}
```

**Comment by Rajinsharwar** [comment]
Thank you so much @swissspidy. I was specifically looking for a function like get_echo(), which I was unaware of its existence. If you can take a look at my changed tests, please. 🙂. Thanks again!

**Review by peterwilsoncc — COMMENTED** [review]
Thanks for adding the unit test.

I think it would be helpful to include one for `orderby` set to a string to ensure that both behaviours are working as expected. 

It would be similar to your existing test but start with:

```php
$_REQUEST['s']       = 'search term';
$_REQUEST['orderby'] = 'title';
```

**Comment by Rajinsharwar** [comment]
Thanks for your suggestion @peterwilsoncc. I have added another test for the single orderby. Let me know if it fits right.

File: tests/phpunit/tests/admin/wpListTable.php [hunk 1]
```diff
@@ -521,4 +521,39 @@ public function data_compat_fields() {
...
+		);
+
+		$output = get_echo( array( $this->list_table, 'search_box' ), array( 'foo Label', 59494 ) );
+
+		$expected_html1 = '<input type="hidden" name="orderby[menu_order]" value="ASC" />';
+		$expected_html2 = '<input type="hidden" name="orderby[title]" value="ASC" />';
+
+		$this->assertStringNotContainsString( $expected_html1, $output );
+		$this->assertStringContainsString( $expected_html2, $output );
+	}
```
Comment (peterwilsoncc) [inline tests/phpunit/tests/admin/wpListTable.php]:
```suggestion
		$this->assertStringContainsString( $expected_html2, $output );
	}

	/**
	 * Tests that "search_box()" works correctly with orderby set to a string.
	 *
	 * @ticket 59494
	 */
	public function test_search_box_works_with_orderby_string() {	
		// Test with one 'orderby' element.
		$_REQUEST['s']       = 'search term';
		$_REQUEST['orderby'] = 'title';

		$output = get_echo( array( $this->list_table, 'search_box' ), array( 'foo Label', 59494 ) );

		$expected_html = '<input type="hidden" name="orderby" value="title" />';

		$this->assertStringContainsString( $expected_html, $output );
	}
```

File: tests/phpunit/tests/admin/wpListTable.php [hunk 2]
```diff
@@ -521,4 +521,39 @@ public function data_compat_fields() {
 			),
 		);
 	}
+
+	/**
+	 * Tests that "search_box()" works correctly with an array of orderby.
+	 *
+	 * @ticket 59494
+	 *
+	 */
```
Comment (peterwilsoncc) [inline tests/phpunit/tests/admin/wpListTable.php]:
```suggestion
	 * @ticket 59494
	 */
```

File: src/wp-admin/includes/class-wp-list-table.php [hunk 3]
```diff
@@ -1868,4 +1879,4 @@ public function _js_vars() {
 
 		printf( "<script type='text/javascript'>list_args = %s;</script>\n", wp_json_encode( $args ) );
 	}
-}
+}
```
Comment (peterwilsoncc) [inline src/wp-admin/includes/class-wp-list-table.php]: Nit: Retain the line break at the end of the file.

```suggestion
}

```

**Review by peterwilsoncc — COMMENTED** [review]
I might have been unclear in my earlier comment.

I was thinking of a test case in which the orderby parameter is set to a string. I've put a suggestion inline. 

I've moved the test to another method as the two tests are for different circumstances and should therefore be tested independently.

**Comment by Rajinsharwar** [comment]
@peterwilsoncc I see. I have committed to the changes. Also, I kept the test for one element array as well. Let me know if this looks good to you. :)

**Review by peterwilsoncc — APPROVED** [review]
Thanks @Rajinsharwar, these tests look good to me.

I took the liberty of pushing a few modifications to the tests:

* splitting out the test for an array of single and multiple values
* renamed `$output` to `$actual` as the latter is more commonly used in the test suite.

Thanks so much for your patience during the review process, I really appreciate the time you took to add the tests.

**Comment by peterwilsoncc** [comment]
Committed in [r58379](https://core.trac.wordpress.org/changeset/58379) / https://github.com/WordPress/wordpress-develop/commit/8cd24bc12c7ac3c588ccbf549b1df42884181fe0

**Pull request closed without merging**